### PR TITLE
Add E2E tests on arm64

### PR DESCRIPTION
**Pull request by kevinzs2048**
Signed-off-by: Kevin Zhao <kevin.zhao@linaro.org>

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by kevinzs2048** [comment]
This PR is regarded with [#14748](https://github.com/etcd-io/etcd/issues/14748)

Just a PR to add e2e test for Arm64 to check if it is green.
If the job is OK, will change it to a periodical job.

**Comment by kevinzs2048** [comment]
@dims @geetasg @chaochn47 first time commit here, so needs maintainers to approve first. Thanks!

**Comment by ahrtr** [comment]
I think we need to add similar tests (see below) for ARM64,
- https://github.com/etcd-io/etcd/blob/main/.github/workflows/tests.yaml
- https://github.com/etcd-io/etcd/blob/main/.github/workflows/e2e.yaml

And remove https://github.com/etcd-io/etcd/blob/main/.github/workflows/functional-arm64.yaml

**Comment by kevinzs2048** [comment]
Hi @ahrtr, thanks!
This one is to add e2e test for Arm64 to verify the function.
After it work, I will add the integration test in the tests.yaml and remove the functional test since it is going to be deprecated.

The reason why it is located to a separate file is that it needs to run on a special Arm64 node and also this will be changed to periodical job.

File: .github/workflows/e2e-arm64.yaml
Comment (serathius) [inline .github/workflows/e2e-arm64.yaml:1]: Instead of adding a separate workflow, can you add arm64 as a job to existing "E2E" workflow?

Comment (ahrtr) [reply .github/workflows/e2e-arm64.yaml:1]: It doesn't work because the ARM machines have different labels. FYI. https://github.com/etcd-io/etcd/pull/15172#discussion_r1084600047

Probably we can add a template to be shared by both the existing E2E workflow and ARM64. But I don't think it's a big deal, It's OK to leave it as it's for now.

**Comment by ahrtr** [comment]
> this will be changed to periodical job.

Why?

**Comment by dims** [comment]
> Why?

@ahrtr the aim is not to be a bottleneck to the current set of CI jobs that are run before we land a PR. The problem is we have only 2 github runners for arm64 and if we have a lot of PRs in flight the time that folks have to wait to land something will be a lot.

**Comment by ahrtr** [comment]
> The problem is we have only 2 github runners for arm64 and if we have a lot of PRs in flight the time that folks have to wait to land something will be a lot.

Makes sense. Thanks @dims for the clarification.

**Review by ahrtr — APPROVED** [review]
LGTM 

Thank you @kevinzs2048 

You might want to change it to periodical job before merging.

**Comment by kevinzs2048** [comment]
@ahrtr Thanks! Modify to periodical job now.

The perviously submit for the e2e-arm64 job is green.

File: .github/workflows/e2e-arm64.yaml
Comment (ahrtr) [inline .github/workflows/e2e-arm64.yaml:4]: 4 hours is too frequent to me. I would suggest to run it daily.

Comment (kevinzs2048) [reply .github/workflows/e2e-arm64.yaml]: sure, change to daily, nightly test at 1 am.

**Comment by ahrtr** [comment]
Please squash the commits.

**Comment by kevinzs2048** [comment]
> Please squash the commits.

Thanks! Done

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks @kevinzs2048

File: .github/workflows/e2e-arm64.yaml
Comment (serathius) [reply .github/workflows/e2e-arm64.yaml:1]: Labels are per job not per forkflow.
```
jobs:
  test:
    runs-on: [Linux, ARM64]
```
You can change it to:

```
jobs:
  amd64:
    runs-on: ubuntu-latest
    ...
  arm64:
    runs-on: [Linux, ARM64]
```

**Review by serathius — APPROVED** [review]

**Pull request merged**
